skin/frontend/base/default/js/oyst/oneclick.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * This file is part of Oyst_OneClick for Magento.
+ *
+ * @license http://www.apache.org/licenses/LICENSE-2.0 Apache License, Version 2.0
+ * @author Oyst <plugin@oyst.com> <@oyst>
+ * @category Oyst
+ * @package Oyst_OneClick
+ * @copyright Copyright (c) 2017 Oyst (http://www.oyst.com)
+ */
+
+/**
+ * OneClick Customs JS
+ */
+
+function ready(fn) {
+    if (document.attachEvent ? 'complete' === document.readyState : 'loading' !== document.readyState) {
+        fn();
+    } else {
+        document.addEventListener('DOMContentLoaded', fn);
+    }
+}
+
+/**
+ * Return the simple product id from a configurable product
+ *
+ * @returns {null}
+ */
+function getSimpleProductId() {
+    if (typeof(spConfig) == 'undefined') {
+        return null;
+    }
+
+    var productCandidates = [];
+
+    spConfig.settings.forEach(function (select, selectIndex) {
+        var attributeId = select.id.replace('attribute', '');
+        var selectedValue = select.options[select.selectedIndex].value;
+
+        spConfig.config.attributes[attributeId].options.forEach(function (option, optionIndex) {
```

Comment:
you could consider using `Array.prototype.filter`
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/filter

Reply:
Ok.